src/goto-programs/goto_sideeffects.cpp

```diff
@@ -287,6 +287,8 @@ void goto_convertt::remove_sideeffects(
       remove_pre(expr, dest, result_is_used);
     else if(statement == "cpp_new" || statement == "cpp_new[]")
       remove_cpp_new(expr, dest, result_is_used);
+    else if(statement == "cpp_delete" || statement == "cpp_delete[]")
```

Comment:
@mikhailramalho I'm not so sure about removing `remove_cpp_delete`. 

If we removed `remove_cpp_delete` and the `else if(statement == "cpp_delete"` branch, it would fail with the error: 
```
Generating GOTO Program
cannot remove side effect (cpp_delete)
```

With `remove_cpp_delete`, we can progress further to generate the GOTO programs

Reply:
Ok, but is the generated goto correct? We are probably failing because the adjust code is missing, have a look at `typecheck_expr_delete`.

I believe we should be calling `convert_cpp_delete` instead.

Reply:
> but is the generated goto correct?

@mikhailramalho the goto generated by old cpp frontend looks about right. 
But the GOTO genrated by new frontend looks wrong. 

> With remove_cpp_delete, we can progress further to generate the GOTO programs

Sorry I should've clarified that I meant the old frontend.